File: sk000801/입문/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = new int[2];
        for (String s : keyinput) {
            if(s.equals("up")) answer[1] += 1;
            if(s.equals("down")) answer[1] += -1;
            if(answer[1] > board[1]/2) answer[1] = board[1]/2;
            if(answer[1] < -board[1]/2) answer[1] = -board[1]/2;
            if(s.equals("left")) answer[0] += -1;
            if(s.equals("right")) answer[0] += 1;
            if(answer[0] > board[0]/2) answer[0] = board[0]/2;
            if(answer[0] < -board[0]/2) answer[0] = -board[0]/2;
        }
        return answer;
    }
}

File: kyw0716/level0/캐릭터의 좌표.js
function solution(keyinput, board) {
  const answer = [0, 0];
  const maxWidth = Math.floor(board[0] / 2);
  const maxHeight = Math.floor(board[1] / 2);

  keyinput.forEach((key) => {
    if (key === "left" && answer[0] - 1 >= -maxWidth) answer[0] -= 1;
    if (key === "right" && answer[0] + 1 <= maxWidth) answer[0] += 1;
    if (key === "down" && answer[1] - 1 >= -maxHeight) answer[1] -= 1;
    if (key === "up" && answer[1] + 1 <= maxHeight) answer[1] += 1;
  });

  return answer;
}

//https://school.programmers.co.kr/learn/courses/30/lessons/120861
